network/f5/bigip_gtm_datacenter.py

```diff
@@ -0,0 +1,569 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: bigip_gtm_datacenter
+short_description: Manage Datacenter configuration in BIG-IP
+description:
+   - Manage BIG-IP data center configuration. A data center defines the location
+     where the physical network components reside, such as the server and link
+     objects that share the same subnet on the network. This module is able to
+     manipulate the data center definitions in a BIG-IP
+version_added: "2.0"
+options:
+  connection:
+    description:
+      - The connection used to interface with the BIG-IP
+    required: false
+    default: rest
+    choices: [ "rest", "icontrol" ]
+  contact:
+    description:
+      - The name of the contact for the data center
+    required: false
+    default: None
+  description:
+    description:
+      - The description of the data center
+    required: false
+    default: None
+  enabled:
+    description:
+      - Whether the data center should be enabled. At least one of state and enabled are required.
+    required: false
+    default: None
+  location:
+    description:
+      - The location of the data center
+    required: false
+    default: None
+  name:
+    description:
+      - The name of the data center
+    required: true
+  password:
+    description:
+      - BIG-IP password
+    required: false
+    default: admin
+  server:
+    description:
+      - BIG-IP host
+    required: true
+  user:
+    description:
+      - BIG-IP username
+    required: false
+    default: admin
+  validate_certs:
+    description:
+      - If C(no), SSL certificates will not be validated. This should only be
+        used on personally controlled sites using self-signed certificates.
+    required: false
+    default: yes
+    choices: [ "yes", "no" ]
+  state:
+    description:
+      - The state of the datacenter on the BIG-IP. When C(present), guarantees
+        that the data center exists. When C(absent) removes the data center
+        from the BIG-IP. C(enabled) will enable the data center and C(disabled)
+        will ensure the data center is disabled. At least one of state and enabled are required.
+    required: false
+    default: None
+    choices: [ "present", "absent" ]
+
+notes:
+   - Requires the bigsuds Python package on the host if using the iControl
+     interface. This is as easy as pip install bigsuds
+
+requirements: [ "bigsuds", "requests" ]
+author: Tim Rupp <caphrim007@gmail.com> (@caphrim007)
+'''
+
+EXAMPLES = """
+- name: Create data center "New York"
+  bigip_gtm_datacenter:
+      server: "big-ip"
+      name: "New York"
+      location: "New York"
+  delegate_to: localhost
+"""
+
+import json
+import socket
+
+try:
+    import bigsuds
+except ImportError:
+    bigsuds_found = False
+else:
+    bigsuds_found = True
+
+try:
+    import requests
+except ImportError:
+    requests_found = False
+else:
+    requests_found = True
```

Comment:
This is never used later. Shouldn't there be an exception if requests isn't found but is attempted to be used?